mne/preprocessing/ieeg/_projection.py

```diff
@@ -0,0 +1,90 @@
+              :ref:`mne flash_bem` using a FLASH scan.
+
+    Parameters
+    ----------
+    %(info_not_none)s Note, ``info`` is modified in place.
+    %(trans_not_none)s
+    %(subject)s
+    %(subjects_dir)s
+    %(picks_base)s only ``ecog`` channels.
+    n_neighbors : int
+        The number of neighbors to use to compute the normal vectors
+        for the projection. Must be 2 or greater. More neighbors makes
+        a normal vector with greater averaging which preserves the grid
+        structure. Fewer neighbors has less averaging which better
+        preserves contours in the grid.
+    %(verbose)s
+    """
+    from scipy.spatial import distance_matrix
+    _validate_type(n_neighbors, int, 'n_neighbors')
+    if n_neighbors < 2:
+        raise ValueError(
+            f'n_neighbors must be 2 or greater, got {n_neighbors}')
+    subjects_dir = get_subjects_dir(subjects_dir, raise_error=True)
+    try:
+        surf = _read_mri_surface(op.join(
+            subjects_dir, subject, 'bem', 'brain.surf'))
+    except FileNotFoundError as err:
+        raise RuntimeError(f'{err}\n\nThe brain surface requires generating '
+                           'a BEM using `mne flash_bem` (if you have '
+                           'the FLASH scan) or `mne watershed_bem` (to '
+                           'use the T1)')
+    # get channel locations
+    picks_idx = _picks_to_idx(info, 'ecog' if picks is None else picks)
+    locs = np.array([info['chs'][idx]['loc'][:3] for idx in picks_idx])
+    trans = _ensure_trans(trans, 'head', 'mri')
+    locs = apply_trans(trans, locs)
+    # compute distances for nearest neighbors
+    dists = distance_matrix(locs, locs)
+    # find angles for brain surface and points
+    angles = _cart_to_sph(locs)
+    surf_angles = _cart_to_sph(surf['rr'])
+    # initialize projected locs
+    proj_locs = np.zeros(locs.shape) * np.nan
+    for i, loc in enumerate(locs):
+        neighbor_pts = locs[np.argsort(dists[i])[:n_neighbors + 1]]
+        normals = list()
+        for pt1, pt2, pt3 in combinations(neighbor_pts, 3):
+            n = np.cross(pt1 - pt2, pt1 - pt3)
+            # flip direction if needed
+            normals.append(n if np.dot(n, loc) >= 0 else -n)
+        normal = np.mean(normals, axis=0)
+        normal /= np.linalg.norm(normal)
+        # find the correct orientation brain surface point nearest the line
+        # https://mathworld.wolfram.com/Point-LineDistance3-Dimensional.html
+        use_rr = surf['rr'][abs(
+            surf_angles[:, 1:] - angles[i, 1:]).sum(axis=1) < np.pi / 4]
+        surf_dists = [np.linalg.norm(np.cross(
+            surf_pt - loc, surf_pt - loc + normal)) for surf_pt in use_rr]
+        proj_locs[i] = use_rr[np.argmin(surf_dists)]
+    # back to the "head" coordinate frame for storing in ``raw``
+    proj_locs = apply_trans(invert_transform(trans), proj_locs)
+    for idx, loc in zip(picks_idx, proj_locs):
+        info['chs'][idx]['loc'][:3] = loc
```

Comment:
you should not need in init file in the tests folder as you should not import anything from there.

Reply:
I was copying off of `nirs` so I'll fix it there as well